### PR TITLE
Fix metadata to accept quoted key table

### LINKED ISSUE #1396: Accept `[package.metadata."docs.rs"]`
I was recently confused by docs.rs not using the provided [configuration](https://docs.rs/about/metadata) from the `Cargo.toml`:
```toml
[package.metadata."docs.rs"]
# Configuration ...
```

Turns out only `[package.metadata.docs.rs]` works (note: no `""`). Not sure why I've written `"docs.rs"` in the first place, but it would be nice to support this too.

### PR DESCRIPTION
**Pull request by chansuke**
Fixes #1396 to accept quoted key table. 

```
[package.metadata."docs.rs"]
# Configuration ...
```

### CODE REVIEW

File: crates/metadata/lib.rs [hunk 1]
```diff
@@ -300,12 +300,19 @@ impl std::str::FromStr for Metadata {
             }
         }
 
-        let table = manifest
+        let plane_table = manifest
```
Comment (jyn514) [inline crates/metadata/lib.rs:303]:
```suggestion
        let plain_table = manifest
```

**Review by jyn514 — APPROVED** [review]
Looks good with the typo fixed :) thanks for the PR!

File: crates/metadata/lib.rs (hunk 1 above)
Comment (chansuke) [reply crates/metadata/lib.rs:303]: @jyn514 Thanks for your review!! I will fix this

**Comment by chansuke** [comment]
Fixed and squashed in one commit

**Review by jyn514 — APPROVED** [review]

**Comment by jyn514** [comment]
This is great, thank you!

**Pull request merged**